src/pages/iou/request/step/IOURequestStepCategory.tsx

```diff
@@ -0,0 +1,171 @@
+    const shouldShowCategory = ReportUtils.isGroupPolicy(report) && (transactionCategory ?? OptionsListUtils.hasEnabledOptions(Object.values(policyCategories ?? {})));
+    const isSplitBill = iouType === CONST.IOU.TYPE.SPLIT;
+    const canEditSplitBill = isSplitBill && reportAction && session?.accountID === reportAction.actorAccountID && TransactionUtils.areRequiredFieldsEmpty(transaction);
+    // eslint-disable-next-line rulesdir/no-negated-variables
+    const shouldShowNotFoundPage = !shouldShowCategory || (isEditing && (isSplitBill ? !canEditSplitBill : !ReportUtils.canEditMoneyRequest(reportAction)));
+
+    const navigateBack = () => {
+        Navigation.goBack(backTo);
+    };
+
+    const updateCategory = (category: ListItem) => {
+        const categorySearchText = category.searchText ?? '';
+        const isSelectedCategory = categorySearchText === transactionCategory;
+        const updatedCategory = isSelectedCategory ? '' : categorySearchText;
+
+        if (transaction) {
+            // In the split flow, when editing we use SPLIT_TRANSACTION_DRAFT to save draft value
+            if (isEditingSplitBill) {
+                IOU.setDraftSplitTransaction(transaction.transactionID, {category: updatedCategory});
+                navigateBack();
+                return;
+            }
+
+            if (isEditing && report) {
+                IOU.updateMoneyRequestCategory(transaction.transactionID, report.reportID, updatedCategory, policy, policyTags, policyCategories);
+                navigateBack();
+                return;
+            }
+        }
+
+        IOU.setMoneyRequestCategory(transactionID, updatedCategory);
+
+        navigateBack();
+    };
+
+    return (
+        <StepScreenWrapper
+            headerTitle={translate('common.category')}
+            onBackButtonPress={navigateBack}
+            shouldShowWrapper
+            shouldShowNotFoundPage={shouldShowNotFoundPage}
+            testID={IOURequestStepCategory.displayName}
+            includeSafeAreaPaddingBottom={false}
+        >
+            <Text style={[styles.ph5, styles.pv3]}>{translate('iou.categorySelection')}</Text>
+            <CategoryPicker
+                selectedCategory={transactionCategory}
+                policyID={report?.policyID ?? ''}
+                onSubmit={updateCategory}
+            />
+        </StepScreenWrapper>
+    );
+}
+
+IOURequestStepCategory.displayName = 'IOURequestStepCategory';
+
+/* eslint-disable rulesdir/no-negated-variables */
+const IOURequestStepCategoryWithWritableReportOrNotFound = withWritableReportOrNotFound(IOURequestStepCategory);
+/* eslint-disable rulesdir/no-negated-variables */
+const IOURequestStepCategoryWithFullTransactionOrNotFound = withFullTransactionOrNotFound(IOURequestStepCategoryWithWritableReportOrNotFound);
+export default withOnyx<IOURequestStepCategoryProps, IOURequestStepCategoryOnyxProps>({
```

Comment:
The order of hocs have changed a(b(c())) -> c(b(a())). This might break the component in very unexpected ways 😅 

[Here](https://github.com/Expensify/App/pull/38995) is how Marcin did it in other PR